nexus/repositories_maven.go

```diff
@@ -0,0 +1,152 @@
+//     but WITHOUT ANY WARRANTY; without even the implied warranty of
+//     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//     GNU Lesser General Public License for more details.
+//
+//     You should have received a copy of the GNU Lesser General Public License
+//     along with Aicura Nexus Client.  If not, see <https://www.gnu.org/licenses/>.
+
+package nexus
+
+import (
+	"context"
+	"encoding/json"
+
+	"golang.org/x/sync/errgroup"
+)
+
+const repositoriesMavenProxyPath = "/repositories/maven/proxy"
+
+// MavenVersionPolicy ...
+type MavenVersionPolicy string
+
+const (
+	// VersionPolicyRelease ...
+	VersionPolicyRelease MavenVersionPolicy = "RELEASE"
+	// VersionPolicyMixed ...
+	VersionPolicyMixed MavenVersionPolicy = "MIXED"
+	// VersionPolicySnapshot ...
+	VersionPolicySnapshot MavenVersionPolicy = "SNAPSHOT"
+)
+
+// MavenLayoutPolicy ...
+type MavenLayoutPolicy string
+
+const (
+	// LayoutPolicyStrict ...
+	LayoutPolicyStrict MavenLayoutPolicy = "STRICT"
+	// LayoutPolicyPermissive ...
+	LayoutPolicyPermissive MavenLayoutPolicy = "PERMISSIVE"
+)
+
+// Maven ...
+type Maven struct {
+	VersionPolicy MavenVersionPolicy `json:"versionPolicy,omitempty"`
+	LayoutPolicy  MavenLayoutPolicy  `json:"layoutPolicy,omitempty"`
+}
+
+// HTTPClient details of how to connect to a Maven Proxy Repository
+type HTTPClient struct {
+	AutoBlock      bool            `json:"autoBlock,omitempty"`
+	Blocked        *bool           `json:"blocked,omitempty"`
+	Connection     *Connection     `json:"connection,omitempty"`
+	Authentication *Authentication `json:"authentication,omitempty"`
+}
+
+// Connection for a HTTPClient
+type Connection struct {
+	Retries                 *int    `json:"retries,omitempty"`
+	UserAgentSuffix         *string `json:"userAgentSuffix,omitempty"`
+	Timeout                 *int32  `json:"timeout,omitempty"`
+	EnableCircularRedirects bool    `json:"enableCircularRedirects,omitempty"`
+	EnableCookies           bool    `json:"enableCookies,omitempty"`
+}
+
+// Authentication for a HTTPClient
+type Authentication struct {
+	Type       string `json:"type,omitempty"`
+	Username   string `json:"username,omitempty"`
+	NtlmHost   string `json:"ntlmHost,omitempty"`
+	NtlmDomain string `json:"ntlmDomain,omitempty"`
+}
+
+// MavenProxyRepository basic structure for a Maven Proxy Repository
+type MavenProxyRepository struct {
+	Repository    `json:",inline"`
+	HTTPClient    HTTPClient    `json:"httpClient,omitempty"`
+	Maven         Maven         `json:"maven,omitempty"`
+	NegativeCache NegativeCache `json:"negativeCache,omitempty"`
+	Proxy         Proxy         `json:"proxy,omitempty"`
+	Storage       Storage       `json:"storage,omitempty"`
+	CleanUp       *CleanUp      `json:"cleanup,omitempty"`
+	RoutingRule   string        `json:"routingRule,omitempty"`
+}
+
+// MavenProxyRepositoryService service to handle all Maven Proxy Repositories operations
+type MavenProxyRepositoryService service
```

Comment:
Same discussion about this matching the repository pattern more closely than a service. Naming will be bitch in this case, unfortunately... But we can discuss naming if you agree with this.